modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts

```diff
@@ -55,13 +55,127 @@ export async function randomCoPrimeTo(x: bigint): Promise<bigint> {
  * @returns {NTilde} The generated NTilde values.
  */
 export async function generateNTilde(bitlength: number): Promise<NTilde> {
-  const ntilde = await generateModulus(bitlength);
+  const { n: ntilde, q1, q2 } = await generateModulus(bitlength);
   const [f1, f2] = await Promise.all([randomCoPrimeTo(ntilde), randomCoPrimeTo(ntilde)]);
   const h1 = modPow(f1, BigInt(2), ntilde);
-  const h2 = modPow(f2, BigInt(2), ntilde);
-  return { ntilde, h1, h2 };
+  const h2 = modPow(h1, f2, ntilde);
+  const beta = modInv(f2, q1 * q2);
+  const [h1wrtH2Proofs, h2wrtH1Proofs] = await Promise.all([
+    generateNTildeProof(
+      {
+        h1: h1,
+        h2: h2,
+        ntilde: ntilde,
+      },
+      f2,
+      q1,
+      q2
+    ),
+    generateNTildeProof(
+      {
+        h1: h2,
+        h2: h1,
+        ntilde: ntilde,
+      },
+      beta,
+      q1,
+      q2
+    ),
+  ]);
+  return {
+    ntilde,
+    h1,
+    h2,
+    ntildeProof: {
+      alpha: h1wrtH2Proofs.alpha.concat(h2wrtH1Proofs.alpha),
+      t: h1wrtH2Proofs.t.concat(h2wrtH1Proofs.t),
+    },
+  };
 }
 
+/**
+ * Generate iterations of Ntilde, h1, h2 discrete log proofs.
+ * @param {NTilde} ntilde Ntilde, h1, h2 to generate the proofs for.
+ * @param {bigint} x Either alpha or beta depending on whether it is a discrete log proof of
+ * h1 w.r.t h2 or h2 w.r.t h1.
+ * @param {bigint} q1 The Sophie Germain prime associated with the first safe prime p1 used to generate Ntilde.
+ * @param {bigint} q2 The Sophie Germain prime associated with the second safe prime p2 used to generate Ntilde.
+ * @param {number} iterations Number of iterations of the ZK proof
+ * (default is 128 as recommened by https://blog.verichains.io/p/vsa-2022-120-multichain-key-extraction)
+ * @returns {NTildeProof} The generated NTilde Proofs.
+ */
+export async function generateNTildeProof(
+  ntilde: NTilde,
```

Comment:
```suggestion
  ntilde: Ntilde,
```

We can defer re-naming these to a future PR